### ISSUE TITLE
Bug for new switch with_wiso in awiesm-2.1-wiso for esm_tools 6.x

### ISSUE THREAD

**Issue by mwerner-awi** [esm-tools/esm_tools#612]
There seems to be a severe bug with the new switch `with_wiso`, which was introduced for awiesm-2.1-wiso in esm_tools 6.x:

My yaml file for running awiesm-2.1-wiso does not contain an entry for `with_wiso` in the block `general`. But I switch off water isotopes with `lwiso: False` entries in both the `echam` and `fesom` block. 

In esm_tools version 5.x, calling esm_runscripts resulted in a correct entry `lwiso = .false.` in the FESOM namelist.config file.
In esm_tools version 6.3.3, calling esm_runscripts results in a **wrong** entry `lwiso = .true.` in the FESOM namelist.config file.

Apparently, the default setting `with_wiso: true` in `esm_tools/configs/setups/awiesm/awiesm-2.1.yaml` "overrides" the setting in my own yaml file. 

**Important to know:** The awiesm-2.1-wiso model runs with this erroneous configuration without any error messages(!). As expected, no water isotope files are generated. However, default ECHAM and FESOM variables differ from a simulation with water isotopes switched on, which must not be the case.

The error was found running awiesm-2.1-wiso on ollie.

**Suggestion:**
When fixing this bug, I suggest a default setting `with_wiso: false` in `esm_tools/configs/setups/awiesm/awiesm-2.1.yaml` for all model versions. Water isotopes should only be included in a simulation if they are explicitly switched on by a user.

**Comment by mandresm** [comment]
Yesterday, while working on awiesm-2.1.yaml improvements I also detected this bug. I am working already in a solution.

**Comment by mandresm** [comment]
@mwerner-awi , I am also doing some cleanup of the wiso switches (`wiso_code` and `with_wiso`). It would be useful for me to have the path of your runscript so that I can test it after I fix the bug and add your suggestions to the yamls. Could you please provide that path?

**Comment by mwerner-awi** [comment]
@mandresm: You can use the following yaml files for testing:
/home/ollie/mwerner/runscripts/awiesm-2.1-wiso/PI_ctrl_awiesm-2.1-wiso_ollie.10nodes.yaml
/home/ollie/mwerner/runscripts/awiesm-2.1-wiso/PI_ctrl_awiesm-2.1-wiso_ollie.10nodes.no_wiso.yaml

**Comment by mandresm** [comment]
Thanks a lot @mwerner-awi. We are working on a bug fix and some cleanups on the `awiesm-2.1.yaml`. These developments are taking place in the branch `feat/awiesm_final`. This will include among other things:
- activating/deactivating wiso using `with_wiso` out-of-the-box, both for the wiso development version `awiesm-2.1-wiso` and standard `awiesm-2.1`
- `with_wiso` will be define `false` as default

This work is part of the outcome of the recent discussion with the AWIESM users. You can read a summary on what was agreed here: https://gitlab.awi.de/paleodyn/Models/namelists/-/issues/5

We expect to finish with the first iteration of the cleanup and bug fixes by the beginning of next week. There, this issue should be solved.

**Comment by mandresm** [comment]
Hi @mwerner-awi,

When testing your runscripts with the new branch that include the fixes I am encountering the following problem with the running of ECHAM. Do you know what the problem might be?

```
  0:  InputInit: Calendar selected: Gregorian
  0:  2001-01-01 00h: Open: jsbach.nc
  0:  2001-01-01 00h: Read: slm (jsbach.nc), avg = .33236
  0:  InputInitialize: mo_input initialized
  0:  new_stream: Adding new stream jsbach
  0:  new_stream: Adding new stream land
  0:  new_stream: Adding new stream accw
  0:  new_stream: Adding new stream veg 
  0:  No time stepping mismatch check defined for MONTHS and YEARS.
  0:  Event <stream   2 event> : interval            1  months : adjustment last : offset[sec]            0   
  0:  new_stream: Adding new stream js_wiso
  0:  new_stream: Adding new stream la_wiso
  0:  new_stream: Adding new stream accw_wiso
  0:  new_stream: Adding new stream yasso
  0:  No time stepping mismatch check defined for MONTHS and YEARS.
  0:  Event <stream   3 event> : interval            1  months : adjustment last : offset[sec]            0   
  0:  
  0: ================================================================================
  0:  
  0: FATAL ERROR in init_lctlib: Error opening landcover library file
  0:  FINISH called from PE: 0
  0:  
  0: --------------------------------------------------------------------------------
```

**Comment by mwerner-awi** [comment]
Hi @mandresm,

No, this error is unfamiliar to me. From the error message it seems that the JSBACH file lctlib.def is not installed properly.

I have just tested a fresh installation of awiesm-2.1.-wiso with the latest esm_tools version (see below) and everything works file. A file named lctlib_nlct21.def is installed in /home/ollie/mwerner/codes/awiesm-2.1-wiso/echam-6.3.05p2-wiso/ and copied/linked as lctlib.def to the working directory when the model is run.

===

mwerner@ollie0:~/esm_tools$ git log
commit f4cd8b1aa92f71b5f9ef9b569d4943eee2deabfe (HEAD -> release, origin/release, origin/HEAD)
Merge: b900ba99 bed84d62
Author: Miguel <63242832+mandresm@users.noreply.github.com>
Date:   Wed Mar 23 08:05:28 2022 +0100

    Merge pull request #615 from esm-tools/fix/find_variable_transformations_and_choose_booleans
    
    Fix/find variable transformations and choose booleans

**Comment by mandresm** [comment]
Hi @mwerner-awi , thanks for the clarification. I think the problem is that I am trying to use as `model_dir` your model directory, and I don't have execution permissions for the file `/home/ollie/mwerner/codes/awiesm-2.1-wiso/echam-6.3.05p2-wiso/lctlib_nlct21.def`. That's why it's not working for me. I will compile my own model and the problem should be solved after that, and I can continue testing.

**Comment by mandresm** [comment]
Hi @mwerner-awi ,

I have managed to run your scripts successfully, making use of the `with_wiso` switch. The branch containing the fixes is `feat/awiesm_final` (#623) which is currently on a testing phase. This branch will be soon merged into `release` and will include the `with_wiso` switch for all the WISO-code versions of AWIESM, plus some additional switches, such as for icebergs, in those versions that support them. We are also cleaning the configuration files, preparing documentation etc, so we make sure, that from now on, AWIESM works as it should, out of the box. Please, feel free to test that branch, and provide any feedback, although bare in mind that that branch is gonna be experiencing changes in the following days of testing.

For this particular issue I ran 4 tests:
- Your `no_wiso` runscript (experiment in `/work/ollie/mandresm/awiesm_final/no_wiso`) - experiment failed in ESM-Tools tidying (I was changing the configuration files, but nothing to worry about)
- Your `with_wiso` runscript (experiment in `/work/ollie/mandresm/awiesm_final/with_wiso`) - experiment failed in ESM-Tools tidying (I was changing the configuration files, but nothing to worry about)
- Your `no_wiso` runscript **simplified** (experiment in `/work/ollie/mandresm/awiesm_final/no_wiso1`) - experiment currently running the second month
- Your `with_wiso` runscript **simplified** (experiment in `/work/ollie/mandresm/awiesm_final/with_wiso1`) - experiment currently running the first month

The simplified versions of the runscripts have many lines commented out. For the lines I commented I have also added valuable comments labelled as `NOTE-MA:`. Please, have a look at those comments if you'd like to get an insight on how to write shorter runscripts. I've checked that all the resulting namelists and namcouple are identical across your original runscripts and the simplified version.

You can find the simplified versions of the runscripts with the comments here:
- `/home/ollie/mandresm/my_scripts/awiesm_final/PI_ctrl_awiesm-2.1-wiso_ollie.10nodes.no_wiso.yaml`
- `/home/ollie/mandresm/my_scripts/awiesm_final/PI_ctrl_awiesm-2.1-wiso_ollie.10nodes.yaml`
